sdks/python/apache_beam/yaml/yaml_provider.py

```diff
@@ -139,6 +139,21 @@ def as_provider_list(name, lst):
   return [as_provider(name, x) for x in lst]
 
 
+def parse_callable_kwargs(spec: Optional[Mapping[str, Any]]):
+  def _parse_callable_kwargs_rec(args, top_level=False):
+    parsed_args = {}
```

Comment:
Nit: Prefere dict comprehensions. This can also make it clearer exactly the pattern you're looking for/handling. 

E.g. 

```
def parse_marked_callables(o):
  if isinstance(o, dict):
    if len(o) == 1 and next(iter(o.keys()) == 'callable'
        return python_callable.PythonCallableWithSource.load_from_source(next(iter(o.values()))
    else:
      return {k: parse_marked_callables(v) for (k, v) in o.items()}
  elif isinstance(o, list):
    return [parse_marked_callables(e) for e in o]
  else:
    return o
```

Reply:
Fixed.